src/app/modules/item/pages/item-forum/item-forum.component.ts

```diff
@@ -0,0 +1,172 @@
+import { Component, Input, OnChanges, OnDestroy, OnInit } from '@angular/core';
+import { ItemData } from '../../services/item-datasource.service';
+import { GroupWatchingService } from '../../../../core/services/group-watching.service';
+import { GetThreadsService } from '../../services/get-threads.service';
+import { ReplaySubject, switchMap, combineLatest, Subject, first } from 'rxjs';
+import { mapToFetchState } from '../../../../shared/operators/state';
+import { distinctUntilChanged, filter, map, startWith, withLatestFrom } from 'rxjs/operators';
+import { ItemRouter } from '../../../../shared/routing/item-router';
+import { ActivatedRoute, NavigationEnd, Router } from '@angular/router';
+import { ItemType } from '../../../../shared/helpers/item-type';
+import { Item } from '../../http-services/get-item-by-id.service';
+
+interface Column {
+  field: string,
+  header: string,
+}
+
+enum ForumTabUrls {
+  MyThreads= '/forum/my-threads',
+  Others = '/forum/others',
+  Group = '/forum/group',
+}
+
+const OPTIONS = [
+  {
+    label: $localize`My threads`,
+    value: ForumTabUrls.MyThreads,
+  },
+  {
+    label: $localize`Other's`,
+    value: ForumTabUrls.Others,
+  },
+];
+
+@Component({
+  selector: 'alg-item-forum',
+  templateUrl: './item-forum.component.html',
+  styleUrls: [ './item-forum.component.scss' ],
+})
+export class ItemForumComponent implements OnInit, OnChanges, OnDestroy {
+  @Input() itemData?: ItemData;
+
+  private readonly url$ = this.router.events.pipe(
+    filter(event => event instanceof NavigationEnd),
+    map(() => this.router.url),
+    startWith(this.router.url),
```

Comment:
that's a bit strange... but I suppose that's just a copy-paste from elsewhere

Reply:
Yes, copy-paste